hazelcast/src/main/java/com/hazelcast/concurrent/atomiclong/AtomicLongService.java

```diff
@@ -147,4 +165,99 @@ private void clearPartitionReplica(int partitionId) {
             }
         }
     }
+
+    @Override
+    public Runnable prepareMergeRunnable() {
+        IPartitionService partitionService = nodeEngine.getPartitionService();
+        Map<Integer, List<AtomicLongContainer>> containerMap = createHashMap(partitionService.getPartitionCount());
```

Comment:
This initial size could be too large:
- there might be more than one member, thus the number of owned partitions is less. You can then count just the owned partitions
- there might be less atomic longs than the number of partitions (e.g. there are 271 partitions but only 5 atomic longs)
- in the future the default partition count will be set to 1999

Reply:
I still have no idea what to put there instead. A normal `HashMap` without an initial size and let it resize? I don't think it's desirable to loop twice, since we could have a huge partition and AtomicLong count, just to get this number right.

Reply:
At the risk of making this complicated with no real benefit (assuming this is only executed during split-brain healing), I would say keep the HashMap default and let it resize. If we insist on optimizing it, and need a heuristic, I would go with the number of containers.

Reply:
I used a default `HashMap`, let's live with the resizing.